src/modules/CurrencyModule.cpp

```diff
@@ -1,46 +1,350 @@
 #include "CurrencyModule.h"
+#include "UmikoBot.h"
+#include "core/Permissions.h"
 
-CurrencyModule::CurrencyModule()
-	: Module("currency", false)
-{
-}
+#include <QtCore/QFile>
+#include <QtCore/QJsonDocument>
+#include <QtCore/qregexp.h>
 
-void CurrencyModule::OnMessage(Discord::Client& client, const Discord::Message& message)
-{
-	m_settings[message.author().id()].currency += 5 + qrand() % 5;
+#define giveawayDeclLoc QString("giveawayAnnouncementId")
+#define currenConfigLoc QString("currencyConfig")
 
-	Module::OnMessage(client, message);
+CurrencyModule::CurrencyModule(UmikoBot* client)
+	: Module("currency", true) {
+
+	m_timer.setInterval(84600 * 1000); //!24hrs = 84600 seconds
+	QObject::connect(&m_timer, &QTimer::timeout, [this, client]() {
+		
+		//!Clear the daily bonus for everyone
+		for (auto it = m_settings.begin(); it != m_settings.end(); ++it) {
+			it->isDailyClaimed = false;
+		}
+
+		if (!isRandomGiveAwayDone) {
+			client->createMessage(giveawayChannelId, "@everyone Today's freebie expires in **10 seconds**. `!claim` it now!");
```

Comment:
I don't think we ever want to tag @everyone or even @here with this bot. And I think 10 seconds is a little low. Especially if we don't tag people. Can we make it 30 or 60 instead and skip the tagging?

Reply:
Aight